server/controllers/posts.js

```diff
@@ -5,6 +5,26 @@ import {
 } from '../validations/posts';
 import joiOptions from '../validations/joiOptions';
 
+const filterPostsQuery = async function filterPostsQuery(type, userId) {
+  let query = {};
+  if (type === 'user-published') {
+    query = {
+      author: userId,
+      public: true,
+      published: true,
```

Comment:
Why not add condition for private posts?

Reply:
done